plugins/node/opentelemetry-instrumentation-knex/src/utils.ts

```diff
@@ -88,3 +88,16 @@ export const extractTableName = (builder: any): string => {
   }
   return table;
 };
+
+export const parseConnectionString = (connectionString: string | undefined) => {
+  if (!connectionString) {
+    return undefined;
+  }
+  const url = new URL(connectionString);
+  return {
+    host: url.hostname,
+    port: parseInt(url.port || '5432', 10),
```

Comment:
I'm interested in this pull request getting merged, so I thought I'd add some notes.

It looks like `connectionString` is for both PostgreSQL and RedShift. It looks like the default port for RedShift is 5439.

I think you can check the URL protocol which should be `postgres:` or `redshift:` and set it to 5432/5439?

```
> (new URL("postgres://user:password@myhost/mydb")).protocol
'postgres:'
```

Reply:
Thanks for the comment. Yes I can do that. However we cannot then cover the case of CockroachDB which uses `postgres` scheme, but has a different default port of 26257.